docs/3.0/develop/blocks.mdx

```diff
@@ -1,161 +1,148 @@
-| [prefect-snowflake](/integrations/prefect-snowflake/)       | Snowflake Credentials      | `snowflake-credentials`                |
-| [prefect-sqlalchemy](/integrations/prefect-sqlalchemy/)   | Database Credentials      | `database-credentials`                 |
-| [prefect-sqlalchemy](/integrations/prefect-sqlalchemy/)   | SQLAlchemy Connector              | `sqlalchemy-connector`                 |
+```python
+from prefect.blocks.core import Block
 
-Anyone can create a custom block type and, optionally, share it with the community.
 
-## Use existing block types
+class Cube(Block):
+    edge_length_inches: float
 
-Blocks are classes that subclass the `Block` base class. You can instantiate and use them like normal classes.
 
-### Instantiate blocks
+# instantiate the type with specific values
+rubiks_cube = Cube(edge_length_inches=2.25)
+
+# store those values in a block document 
+# on the server for future use
+rubiks_cube.save("rubiks-cube")
+
+
+# instantiate and save another block document
+tiny_cube = Cube(edge_length_inches=0.001)
+tiny_cube.save("tiny")
+```
+
+Block documents can also be created and updated in the UI or API for easy change management. 
+This allows you to work with slowly changing configuration without having to redeploy all workflows that rely on it; for example, you may use this to rotate credentials on a regular basis without touching your deployments.
+
+<Warning>
+When using the generic `Block.load` interface for block documents, the Python class that created the type _must_ be importable.
+For this reason, we highly recommend loading block documents by first importing the block class and then calling its `load` method with the relevant document name.
+</Warning>
+
+### Block class
+
+A block _class_ is the primary user-facing object; it is a Python class whose attributes are loaded from a block document. Most Prefect blocks encapsulate additional functionality built on top of the block document. 
 
-To instantiate a block that stores an S3 bucket value, use the `S3Bucket` block:
+For example, an `S3` block contains methods for downloading data from, or upload data to, an S3 bucket; a `SnowflakeConnector` block contains methods for querying Snowflake databases.
+
+Returning to our `Cube` example from above:
 
 ```python
-from prefect_aws.s3 import S3Bucket
+from prefect.blocks.core import Block
 
-bucket_block = S3Bucket(bucket_name="data-bucket")
+class Cube(Block):
+    edge_length_inches: float
 
+    def get_volume(self):
+        return self.edge_length_inches ** 3
+
+    def get_surface_area(self):
+        return 6 * self.edge_length_inches ** 2
+
+
+rubiks_cube = Cube.load("rubiks-cube")
+rubiks_cube.get_volume() # 11.390625
 ```
 
-### Save blocks
+<Warning>
+It is very important to note that the class itself is _not_ stored server-side when registering block types and block documents. 
```

Comment:
```suggestion
The class itself is _not_ stored server-side when registering block types and block documents. 
```

Reply:
I feel like putting this statement in a Warning shows its importance, but feel free to keep the more verbose version if you like.